adafruit_as7341.py

```diff
@@ -43,69 +43,89 @@
 from adafruit_register.i2c_bit import RWBit
 from adafruit_register.i2c_bits import ROBits, RWBits
 
-_AS7341_DEVICE_ID = const(0b001001)  # Correct content of WHO_AM_I register
-_AS7341_I2CADDR_DEFAULT = const(0x39)  # AS7341 default i2c address
-_AS7341_CHIP_ID = const(0x09)  # AS7341 default device id from WHOAMI
-_AS7341_WHOAMI = const(0x92)  # Chip ID register
-_AS7341_CONFIG = const(0x70)  # Enables LED control and sets light sensing mode
-_AS7341_GPIO = const(0x73)  # Connects photo diode to GPIO or INT pins
-_AS7341_LED = const(0x74)  # LED Register; Enables and sets current limit
-_AS7341_ENABLE = const(
+try:
+    from typing import Tuple, Optional, Any, Callable, TypeVar
+
+    # Only needed for typing
+    import busio  # pylint: disable=unused-import
+
+    TCallable = TypeVar("TCallable", bound=Callable[..., Any])
+
+except ImportError:
+    pass
+
+# Correct content of WHO_AM_I register
+_AS7341_DEVICE_ID: int = const(0b001001)
```

Comment:
@tammymakesthings @tekktrik Do either of you have this device to test with? 

For some reason I was thinking that either mpy-cross or the CircuitPython microcontrollers had trouble with typed variables like these (as opposed to function arguments / returns). I thought we ran into this issue on a different library during the typing PR but I'm not certain which one. 

If possible I think it would be good to do a quick test with the simpletest in this repo to ensure the code from this branch runs on microcontroller without issue.

Reply:
I don't have an AS7341 board. :-(

Reply:
I don't either unfortunately, though it does seem like a neat little breakout board!  Happy to get one on order, but maybe we should see if anyone on the Discord can test?

Reply:
I'm away from my hardware now, but this trick should work for any library that uses `const()` right?

Reply:
I did just try running `mpy-cross` on the .py file manually and didn't see any issues. I also copied the `.py` file into the `lib/` folder on a couple of CircuitPython devices I have lying on my desk, and didn't see any issues. I'm not sure what other testing I can attempt without an actual AS7341 board.